geoapps/io/constants.py

```diff
@@ -0,0 +1,143 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+valid_parameters = [
+    "data",
+    "out_group",
+    "workspace",
+    "save_to_geoh5",
+    "inversion_type",
+    "inversion_style",
+    "forward_only",
+    "inducing_field_aid",
+    "core_cell_size",
+    "octree_levels_topo",
+    "octree_levels_obs",
+    "octree_levels_padding",
+    "depth_core",
+    "max_distance",
+    "padding_distance",
+    "chi_factor",
+    "max_iterations",
+    "max_cg_iterations",
+    "max_global_iterations",
+    "n_cpu",
+    "max_ram",
+    "initial_beta_ratio",
+    "tol_cg",
+    "ignore_values",
+    "no_data_value",
+    "resolution",
+    "window",
+    "alphas",
+    "reference_model",
+    "starting_model",
+    "model_norms",
+    "data",
+    "uncertainty_mode",
+    "receivers_offset",
+    "topography",
+    "result_folder",
+    "detrend",
+    "data_file",
+    "new_uncert",
+    "input_mesh",
+    "save_to_geoh5",
+    "input_mesh_file",
+    "inversion_mesh_type",
+    "shift_mesh_z0",
+    "receivers_offset",
+    "gradient_type",
+    "initial_beta",
+    "initial_beta_ratio",
+    "lower_bound",
+    "upper_bound",
+    "max_distance",
+    "max_chunk_size",
+    "chunk_by_rows",
+    "output_tile_files",
+    "no_data_value",
+    "parallelized",
+]
+
+required_parameters = ["inversion_type", "core_cell_size"]
```

Comment:
`required_parameters = ["inversion_type", ]`

Reply:
Of course, since we're moving to providing the mesh!